source/cursorManager.py

```diff
@@ -27,6 +26,10 @@
 import ui
 from textInfos import DocumentWithPageTurns
 
+# search history list constants
+SEARCH_HISTORY_FRONT = 0
+SEARCH_HISTORY_MAX_SIZE=20

```

Comment:
I think it would be much clearer to use `SEARCH_HISTORY_FIRST_INDEX = 0` and `SEARCH_HISTORY_LAST_INDEX = 19`. From this it's obvious there are 20 elements, there is no conversion required between size and final index eg: `del entries[(SEARCH_HISTORY_MAX_SIZE -1):]`, it's clear we are talking about indices rather than values.

While in C++ it's a common pattern to define the size as a constant, this is typically because you want to pre-allocate your data structure. This is not necessary here, and in fact `SEARCH_HISTORY_MAX_SIZE` is never used directly at all.

Reply:
> 
> 
> I think it would be much clearer to use `SEARCH_HISTORY_FIRST_INDEX = 0` and `SEARCH_HISTORY_LAST_INDEX = 19`. From this it's obvious there are 20 elements, there is no conversion required between size and final index eg: `del entries[(SEARCH_HISTORY_MAX_SIZE -1):]`, it's clear we are talking about indices rather than values.

This is a subjective matter. I have chosen to use constants naming the "business" representation ratter than the computational representation.
From an algorithm point of view, I am inserting items at the front of the list and I am checking if the length of my list is greater than the maximum amount of items allowed for that list, and this is why the SEARCH_HISTORY_MAX_SIZE exists. May be I should have used SEARCH_HISTORY_MAX_LENGTH instead, in Portuguese these two words translate to the same word and this is why I had that mistake..
If the list is now greater or equals than the maximum length of items, 20 items (not 19) then I truncate it to 19 items. And I do so because right at the next line I will insert the 20th item at the front of the list.
In fact, the only function having to deal with indexes, which play the computation not the business roles, is _truncateSearchHistory. Because it has to deal with computational requirements, such as converting a length one wants to truncate the search history to a index needed by python, its logic has been separated and encapsulated on a private method. The rule is that, at the moment, it is necessary to convert a length (business realm) to an index (computation realm) by subtracting one from the desired length because python is zero indexing collections. If the maximum length changes, the corresponding index must be equals to the new length minus one.

> 
> While in C++ it's a common pattern to define the size as a constant, this is typically because you want to pre-allocate your data structure. This is not necessary here, and in fact `SEARCH_HISTORY_MAX_SIZE` is never used directly at all.


It is directly used in line 88:
if len(searchEntries) >= SEARCH_HISTORY_MAX_SIZE:

Notice that except by the word size instead of length this is basically raw English: if length of searchEntris is greater than or equals to the maximum allowed length for this list then ...
truncate the list so that it is the size it should be minus one ... and then
add the current term to the front of the list

So I would define the constants exactly the way they are. However, you are the reviewer, not me and, to be franc, users will benefit from the search history, not from this little discussion. although I myself appreciate it because it is always good to talk to and have code reviewed by talented programmers.

I will therefore implement the suggested changes and commit them possibly in one day.